src/controllers/auth.js

```diff
@@ -24,11 +58,15 @@ class AuthController extends BaseController {
    * @param  {string}   token     The jwt token to check
    * @return {Promise|*|PromiseLike<T>|Promise<T>}
    */
-  checkToken (token) {
+  checkToken (token = undefined) {
+    if (token === undefined && this.authenticationToken) {
```

Comment:
(to be discussed) Personnally, I would throw if no token is defined or, perhaps better, I would let Kuzzle respond that it needs a token to be verified (like it was done before your changes)

And if users want to verify the stored token, they only need to use it like this: `kuzzle.auth.checkToken(kuzzle.jwt)`
Don't forget that the SDK automatically does an auth:checkToken on the stored jwt when a network connection is (re-)established.

Reply:
I know this was the original behaviour but most of the time users want to check if the token stored by the SDK is still valid so IMHO they will understand that if you call this method without a specific token, it will check the SDK internal token.
Also it's more simpler to get ride of the `kuzzle.jwt` argument.
But maybe I can throw an error if `checkToken` is called without a token and if there is no internal token

Reply:
The token is already automatically checked by the SDK, but... why not In that case, you have to document it properly in the checkToken SDK documentation